generators/base/generator/config/authority.mts

```diff
@@ -0,0 +1,32 @@
+/**
+ * Copyright 2013-2022 the original author or authors from the JHipster project.
+ *
+ * This file is part of the JHipster project, see https://www.jhipster.tech/
+ * for more information.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { OAUTH2 } from '../../../../jdl/jhipster/authentication-types.js';
+import databaseTypes from '../../../../jdl/jhipster/database-types.js';
+
+const { COUCHBASE, MONGODB, NEO4J, SQL, NO: NO_DATABASE } = databaseTypes;
+
+const isUsingBuiltInAuthorityConfiguration = (config: any): boolean => {
+  return (
+    config ||
+    (!config.skipUserManagement && [SQL, MONGODB, COUCHBASE, NEO4J].includes(config.databaseType)) ||
+    (config.authenticationType === OAUTH2 && config.databaseType !== NO_DATABASE)
```

Comment:
These conditions should be changed:

`skipUserManagement` should be split into something like `generateUserManagement` and `generateUserModel`.
It's quite confusing that oauth2 forces `skipUserManagement`, but adds a lot of workarounds to generate the user model for relationships.